kratos/sources/model_part_io.cpp

```diff
@@ -138,10 +131,11 @@ std::size_t ModelPartIO::ReadNodesNumber()
 
 void ModelPartIO::WriteNodes(NodesContainerType const& rThisNodes)
 {
-    (*mpStream) << "Begin Nodes" << std::endl;
-    for(NodesContainerType::const_iterator it_node = rThisNodes.begin() ; it_node != rThisNodes.end() ; it_node++)
-        (*mpStream) << "\t" << it_node->Id() << "\t" << it_node->X()  << "\t" << it_node->Y() << "\t" << it_node->Z() << std::endl;
-    (*mpStream) << "End Nodes" << std::endl << std::endl;
+    mpStream->precision(10);
+    (*mpStream) << "Begin Nodes\n" << std::scientific;
+    for(auto it_node = rThisNodes.begin() ; it_node != rThisNodes.end() ; it_node++)
```

Comment:
```suggestion
    for(NodesContainerType::const_iterator it_node = rThisNodes.begin() ; it_node != rThisNodes.end() ; ++it_node)
```

I am pretty sure that `auto` will result in a regular and not a const iterator!